GPU/Common/FragmentShaderGenerator.cpp

```diff
@@ -1078,16 +1082,37 @@ bool GenerateFragmentShader(const FShaderID &id, char *buffer, const ShaderLangu
 		return false;
 	}
 
-	// Final color computed - apply color write mask.
-	// TODO: Maybe optimize to only do math on the affected channels?
-	// Or .. meh. That would require more shader bits. Though we could
-	// of course optimize for the common mask 0xF00000, though again, blue-to-alpha
-	// does a better job with that.
-	if (colorWriteMask) {
+	// Final color computed - apply logic ops and bitwise color write mask, through shader blending, if specified.
+	if (colorWriteMask || replaceLogicOp) {
```

Comment:
`if (colorWriteMask && !hasPackUnorm4x8) {` above needs to do this now for `replaceLogicOp` too.

-[Unknown]

Reply:
Good catch, I think that's what the tests caught, too.